diesel/src/mysql_like/backend.rs

```diff
@@ -0,0 +1,26 @@
+//! The MySQL backend
+use byteorder::NativeEndian;
+
+use backend::*;
+use sql_types::TypeMetadata;
+
+pub trait MysqlLikeBackend {
```

Comment:
I would make that trait only a marker trait (== it does not have any items)

Then both, the mysql and the mariadb backend implement the corresponding traits (`Backend` and `TypedMetadata`). Additionally `MysqlLike` is implemented, similar to the `Support*` traits.

Reply:
That's what I tried initially, and adding those associated types was some experimentation. I'll be glad to get rid of them after trying the suggestions in the previous comment.

I'll also make a better branch name and make the PR into the diesel repo.